packages/permissionless/accounts/kernel/utils/getActualKernelVersion.ts

```diff
@@ -0,0 +1,67 @@
+import { type Address, type Client, getContract } from "viem"
+import type { KernelVersion } from "../toEcdsaKernelSmartAccount.js"
+
+// If the kernel contract is already deployed, we can get the actual version from the contract
+export async function getActualKernelVersion(
+    client: Client,
+    address: Address,
+    version: KernelVersion<"0.6" | "0.7">
+): Promise<KernelVersion<"0.6" | "0.7">> {
+    try {
+        const contract = getContract({
+            address,
+            abi: [
```

Comment:
I'd rather define abi as a constant variable for reusability.
`const EIP712_DOMAIN_ABI = [
    {
        type: "function",
        name: "eip712Domain",
        inputs: [],
        outputs: [
            { name: "fields", type: "bytes1", internalType: "bytes1" },
            { name: "name", type: "string", internalType: "string" },
            { name: "version", type: "string", internalType: "string" },
            { name: "chainId", type: "uint256", internalType: "uint256" },
            { name: "verifyingContract", type: "address", internalType: "address" },
            { name: "salt", type: "bytes32", internalType: "bytes32" },
            { name: "extensions", type: "uint256[]", internalType: "uint256[]" },
        ],
        stateMutability: "view",
    },
];`